indico/modules/events/layout/controllers/images.py

```diff
@@ -56,24 +59,36 @@ def _process(self):
         files = request.files.getlist('image')
         num = 0
         for f in files:
+
             filename = secure_filename(f.filename, 'image')
+            mtype = guess_type(filename)[0]
+            if mtype not in config.ALLOWED_IMAGE_MIME_TYPES:
+                continue
             data = BytesIO()
             shutil.copyfileobj(f, data)
             data.seek(0)
-            try:
-                image_type = Image.open(data).format.lower()
-            except IOError:
-                # Invalid image data
-                continue
+            if mtype == 'application/pdf':
+                try:
+                    _pdf = PdfFileReader(data)
+                    image_type = 'pdf'
+                    content_type = mtype
+                except Exception:
+                    continue
+            else:
+                try:
+                    image_type = Image.open(data).format.lower()
```

Comment:
ah, didn't see this when posting the comment above regarding getting the image type properly